net_sink.go

```diff
@@ -312,27 +317,41 @@ func (s *netSink) run() {
 			// Drop through in case retryc has nothing.
 		}
 
+		// if the channel is at capacity, it's blocking, let's signal for it to be flushed and unblock
+		if len(s.outc) == cap(s.outc) {
+			s.Flush()
+		}
+
 		select {
-		case <-t.C:
-			s.flush()
 		case done := <-s.doFlush:
-			// Only flush pending buffers, this prevents an issue where
```

Comment:
this case was effectively redundant, competing with `case buf := <-s.outc` to process and clear outc . repurposing it to handle batching